pkg/bucket/bandwidth/measurement.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bandwidth
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	// betaBucket is the weight used to calculate exponential moving average
+	betaBucket = 0.1 // Number of averages considered = 1/(1-betaObject)
+)
+
+// bucketMeasurement captures the bandwidth details for one bucket
+type bucketMeasurement struct {
+	lock                       sync.Mutex
+	cumulativeBytes            uint64        // Sum of all bytes transferred before the latest update
+	cumulativeTime             time.Duration // Total time taken for bytes
+	sumOfTimePerObject         time.Duration // Sum of time taken by objects in a window
+	sumOfTimePerObjectInWindow time.Duration // Sum of time taken by objects in a window
```

Comment:
Same comment. I assume this means something different.

Reply:
Response below.